Automattic/jetpack
# Backup: Factor out backups api fetch into hook

Factors out a bunch of fetching code into a new hook file 


#### Changes proposed in this Pull Request:

* Create a new hook that's able to fetch data for the component.
* Factors out fetching into new gook

#### Other information:



#### Jetpack product discussion
None

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

**Note: The final diff looks weird as GitHub tries to match some lines in weird ways. It's better to look at each commit to understand what's being done.**

1. Launch a [JN site with Backup standalone and this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-backup=update/backup-move-fetch-hook-into-hooks&nojetpack&wp-debug-log)
2. Keep the JS console open and expect to see no warning for the following steps
3. Connect, expect to see the in progress status
4. Wait until backup is complete. Expect to see information about the backup.
